### ISSUE TITLE
New external pods not propagated in mesh

### ISSUE THREAD

**Issue by arne-klein** [istio/istio#12371]
**Describe the bug**
Sometimes an update of a deployment outside of the mesh, which triggers the creation of a new pod and removal of the old pod, will not propagate the changed IP of the service/pod to the mesh, even though the mesh status is displayed as in sync.

The updated pod is not a part of the mesh / does not have an Istio sidecar.
This happened for us several times for a deployment in the kube-system namespace. Even removal and recreation of pod and service did not fix the problem.

Pod in question
```
NAME                                                             READY   STATUS    RESTARTS   AGE     IP              NODE                                                  NOMINATED NODE
apm-server-5f7c977dcf-pvdbq                                      1/1     Running   0          44m     10.208.132.54   gke-gke-west1-blue-gke-west1-blue-np1-7eb0e730-wh46   <none>
```
Service
```
NAME                   TYPE        CLUSTER-IP       EXTERNAL-IP   PORT(S)             AGE    SELECTOR
apm-server             ClusterIP   10.208.195.101   <none>        8200/TCP            3d7h   app=apm-server
```
Pilot discovery log (nothing more relevant, no EDS push afterwards).
```
2019-03-08T21:52:51.178342Z     info    Handling event update for pod apm-server-5f7c977dcf-pvdbq in namespace kube-system -> 10.208.132.54
2019-03-08T21:52:51.228698Z     info    Handling event update for pod apm-server-7588c7f76-p7m7v in namespace kube-system -> 10.208.134.16
2019-03-08T21:52:57.116826Z     info    Handling event update for pod apm-server-7588c7f76-p7m7v in namespace kube-system -> 10.208.134.16
2019-03-08T21:53:00.329555Z     info    Handling event update for pod apm-server-7588c7f76-p7m7v in namespace kube-system -> 10.208.134.16
2019-03-08T21:53:00.340591Z     info    Handling event delete for pod apm-server-7588c7f76-p7m7v in namespace kube-system -> 10.208.134.16
2019-03-08T21:53:05.844839Z     error   Failed to update lock: Operation cannot be fulfilled on configmaps "istio-ingress-controller-leader-istio": the object has been modified; please apply your changes to the latest version and try again
2019-03-08T21:53:15.863596Z     error   Failed to update lock: Operation cannot be fulfilled on configmaps "istio-ingress-controller-leader-istio": the object has been modified; please apply your changes to the latest version and try again
2019-03-08T21:53:25.881770Z     error   Failed to update lock: Operation cannot be fulfilled on configmaps "istio-ingress-controller-leader-istio": the object has been modified; please apply your changes to the latest version and try again
```
`istioctl proxy-status`: everything in sync. A part of the output including the ingressgateways:
```
istio-ingressgateway-74c8749c8b-4ngg7.istio-system          SYNCED     SYNCED     SYNCED (100%)     SYNCED     istio-pilot-55bbcdc666-mh8sd     1.0.2
istio-ingressgateway-74c8749c8b-pkpwd.istio-system          SYNCED     SYNCED     SYNCED (100%)     SYNCED     istio-pilot-55bbcdc666-mh8sd     1.0.2
istio-ingressgateway-74c8749c8b-q4fzr.istio-system          SYNCED     SYNCED     SYNCED (100%)     SYNCED     istio-pilot-55bbcdc666-5lr5v     1.0.2
istio-ingressgateway-prod-6cd97bdd99-5rtrb.istio-system     SYNCED     SYNCED     SYNCED (100%)     SYNCED     istio-pilot-55bbcdc666-5lr5v     1.0.2
istio-ingressgateway-prod-6cd97bdd99-c4st2.istio-system     SYNCED     SYNCED     SYNCED (100%)     SYNCED     istio-pilot-55bbcdc666-mh8sd     1.0.2
istio-ingressgateway-prod-6cd97bdd99-tn79v.istio-system     SYNCED     SYNCED     SYNCED (100%)     SYNCED     istio-pilot-55bbcdc666-mh8sd     1.0.2
```
`istioctl proxy-config endpoints {anyServiceInMesh} | grep apm-server` only shows the old IP of the service, the new one is missing.
```
10.208.134.16:8200      HEALTHY     outbound|8200||apm-server.kube-system.svc.cluster.local
```

**Expected behavior**
The current state of available pods/IPs per service to be correctly distributed in the mesh.

**Steps to reproduce the bug**
Sometimes update of the deployment as described above.

**Version**
Istio v1.0.6
```
Version: 1.0.6
GitRevision: 98598f88f6ee9c1e6b3f03b652d8e0e3cd114fa2
User: root@464fc845-2bf8-11e9-b805-0a580a2c0506
Hub: docker.io/istio
GolangVersion: go1.10.4
BuildStatus: Clean
```

```
Client Version: version.Info{Major:"1", Minor:"13", GitVersion:"v1.13.3", GitCommit:"721bfa751924da8d1680787490c54b9179b1fed0", GitTreeState:"clean", BuildDate:"2019-02-01T20:08:12Z", GoVersion:"go1.11.5", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"12+", GitVersion:"v1.12.5-gke.5", GitCommit:"2c44750044d8aeeb6b51386ddb9c274ff0beb50b", GitTreeState:"clean", BuildDate:"2019-02-01T23:53:25Z", GoVersion:"go1.10.8b4", Compiler:"gc", Platform:"linux/amd64"}
```

**Installation**
With the helm chart, using `template` (not tiller).

**Environment**
GKE

**Cluster state**
Not attached, as it is a production cluster. More information can be provided as needed, a cluster state dump is saved.

**Comment by Stono** [comment]
Is this another case of using incremental EDS and kube-system namespace services?  
See: https://github.com/istio/istio/issues/11658 and https://github.com/istio/istio/pull/12028

**Comment by arne-klein** [comment]
Thanks! It seems to be the same underlying issue as in those two reports, which is being fixed in #12028 and #12030.
Hoping for the 1.0.7 release #12090 soon to resolve the issue.
